### PR TITLE
Added flags for auto-vectorization and simple tests

### PR DESCRIPTION
**Pull request by ITO444**
Flags added:
-vectorize (Enables vectorizer, currently does nothing)
-no-vectorize (Disables vectorizer, currently does nothing)
-dvectorize (dumps extra information about vectorization if vectorizer is enabled, currently only counts number of instructions)

Only the skeleton is created, the pass to the vectorizer currently does not change the cfg.

Tests compile and run very a simple function (adding 2 pairs of numbers) implemented normally (compiled with and without the vectorizer) and implemented using intrinsics then compares their outputs.

### CODE REVIEW

**Review by gretay-js — CHANGES_REQUESTED** [review]
Looks good, just a few small suggestions in the comments.
Please add the new files `vectorize.ml{,i}` to the list in `backend/.ocamlformat-enable`.

File: tests/backend/vectorizer/dune [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  (and
+   (= %{context_name} "main")
+   (<> %{system} macosx)))
+ (targets test1.out test1_simd.out test1_vectorized.out)
+ (deps test_runner1.exe test_runner1_simd.exe test_runner1_vectorized.exe)
+ (action
+  (progn
+   (with-outputs-to
+    grep_test1_vectorizer_cmx.out
+    (run grep "*** Vectorization" test1.cmx.dump --line-regexp --max-count=1))
```
[the remark this answers is not in the document]
Comment (ITO444) [reply tests/backend/vectorizer/dune]: done

File: tests/backend/vectorizer/test1_simd.ml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+end
+
+let add_pairs (a0, a1 : int64 * int64) (b0, b1 : int64 * int64) =
+  let sum_vector = (Int64x2.add (Int64x2.of_int64s a0 a1) (Int64x2.of_int64s b0 b1)) in
+  (Int64x2.low_int64 sum_vector, Int64x2.high_int64 sum_vector)
+;;
+
+let () =
+  let sum0, sum1 = add_pairs (0L, 1L) (2L, 3L) in
+  Printf.printf "%Lx %Lx\n" sum0 sum1
```
[the remark this answers is not in the document]
Comment (ITO444) [reply tests/backend/vectorizer/test1_simd.ml:59]: done

File: tests/backend/vectorizer/test1_simd.ml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    -> t
+    = "ocaml_simd_unreachable" "caml_sse_vec128_low_64_to_high_64"
+  [@@noalloc] [@@unboxed] [@@builtin]
+
+  let[@inline always] set a b =
+    (* movq, + insert -> 3 cycle latency
+      this           -> 2 cycle latency *)
+    let a = low_of a in
+    let b = low_of b in
+    low_64_to_high_64 a b
```
[the remark this answers is not in the document]
Comment (ITO444) [reply tests/backend/vectorizer/test1_simd.ml]: done

**Review by gretay-js — APPROVED** [review]

**Pull request merged**